src/main/java/org/reaktivity/nukleus/socks/internal/stream/AbstractStreamProcessor.java

```diff
@@ -0,0 +1,383 @@
+        {
+            DataFW dataFW = context.dataRW
+                .wrap(context.writeBuffer, 0, context.writeBuffer.capacity())
+                .streamId(targetStreamId)
+                .payload(p -> p.set(
+                    flyweight.buffer(),
+                    flyweight.offset() + attemptOffset,
+                    targetBytes))
+                .build();
+            target.accept(
+                dataFW.typeId(),
+                dataFW.buffer(),
+                dataFW.offset(),
+                dataFW.sizeof());
+            senderPartial.update(targetCredit);
+            attemptOffset += targetBytes;
+        }
+    }
+
+    protected void doForwardData(
+        OctetsFW payload,
+        long streamId,
+        MessageConsumer streamEndpoint)
+    {
+        doForwardData(
+            payload,
+            payload.sizeof(),
+            streamId,
+            streamEndpoint);
+    }
+
+    protected void doForwardData(
+        OctetsFW payload,
+        int payloadLength,
+        long streamId,
+        MessageConsumer streamEndpoint)
+    {
+        doForwardData(
+            payload.buffer(),
+            payload.offset(),
+            payloadLength,
+            streamId,
+            streamEndpoint);
+    }
+
+    protected void doForwardData(
+        DirectBuffer payloadBuffer,
+        int payloadOffset,
+        int payloadLength,
+        long streamId,
+        MessageConsumer streamEndpoint)
+    {
+        DataFW dataForwardFW = context.dataRW.wrap(context.writeBuffer, 0, context.writeBuffer.capacity())
+            .streamId(streamId)
+            .payload(p -> p.set(
+                payloadBuffer,
+                payloadOffset,
+                payloadLength))
+            .extension(e -> e.reset())
+            .build();
+        streamEndpoint.accept(
+            dataForwardFW.typeId(),
+            dataForwardFW.buffer(),
+            dataForwardFW.offset(),
+            dataForwardFW.sizeof());
+    }
+
+    protected void handleHighLevelDataBufferUnwind(
+        DataFW data,
+        int currentTargetCredit,
+        long targetStreamId,
+        MessageConsumer sourceThrottle,
+        long sourceStreamId,
+        FragmentedForwarderPartial forwarderPartial,
+        FragmentedForwarderComplete forwarderComplete)
+    {
+        OctetsFW payload = data.payload();
+        if (this.slotWriteOffset != 0)
+        {
+            MutableDirectBuffer acceptBuffer = context.bufferPool.buffer(this.slotIndex, this.slotWriteOffset);
+            acceptBuffer.putBytes(
+                0,
+                payload.buffer(),
+                payload.offset(),
+                payload.sizeof());
+            slotWriteOffset += payload.sizeof();
+            return;
+        }
+
+        if (currentTargetCredit >= payload.sizeof())
+        {
+            try
+            {
+                forwarderComplete.updateSentFullData(payload);
+            }
+            catch (Exception e)
```

Comment:
Please see my previous comment about avoiding try...catch. This should be removed.